diff -ruN gdb-9.1/gdb/gdbsupport/pathstuff.c gdb-9.1-mingw/gdb/gdbsupport/pathstuff.c
--- gdb-9.1/gdb/gdbsupport/pathstuff.c	2020-02-08 12:50:14 +0000
+++ gdb-9.1-mingw/gdb/gdbsupport/pathstuff.c	2020-08-26 14:31:06 +0000
@@ -231,6 +231,15 @@
 #endif
 
   const char *home = getenv ("HOME");
+
+#ifdef WIN32
+  if (home == nullptr)
+    {
+      /* This is used when gdb was compiled in static mode on Windows.  */		
+      home = getenv ("USERPROFILE");
+    }
+#endif
+  
   if (home != NULL)
     {
       /* Make sure the path is absolute and tilde-expanded.  */
